File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/constants/Endpoints.java
package com.emerchantpay.gateway.genesisandroid.api.constants;

import java.io.Serializable;

public class Endpoints implements Serializable {

	private String endpointName;

	// Domain for E-ComProcessing's Genesis instance
	public static Endpoints ECOMPROCESSING = new Endpoints("e-comprocessing.net");

	// Domain for Emerchantpay's Genesis instance
	public static Endpoints EMERCHANTPAY = new Endpoints("emerchantpay.net");

	public Endpoints(String endpointName) {
		this.endpointName = endpointName;
	}

	public String getEndpointName() {
		return this.endpointName;
	}

	public String toString() {
		return getEndpointName();
	}
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/constants/Endpoints.kt
package com.emerchantpay.gateway.genesisandroid.api.constants

enum class Endpoints(val endpointName: String) {
    // Domain for E-ComProcessing's Genesis instance
    ECOMPROCESSING("e-comprocessing.net"),

    // Domain for Emerchantpay's Genesis instance
    EMERCHANTPAY("emerchantpay.net"),

    // Domains for Consumer API
    RETRIEVE_CONSUMER_EMERCHANTPAY("emerchantpay.net/v1/retrieve_consumer"),
    RETRIEVE_CONSUMER_ECOMPROCESSING("e-comprocessing.net/v1/retrieve_consumer");


    override fun toString(): String {
        return endpointName
    }
}